extensions/amp-lightbox-gallery/1.0/base-element.js

```diff
@@ -102,16 +108,42 @@ export class BaseElement extends PreactBaseElement {
  * @return {!Array<PreactDef.Renderable>}
  */
 function getLightboxElements(document, open) {
-  return toArray(document.querySelectorAll(LIGHTBOX_ELIGIBLE_SELECTORS)).map(
-    (element, index) => {
-      element.addEventListener('click', () => open(index));
-      return (
-        <WithLightbox
-          render={() => <img srcset={srcsetFromElement(element).stringify()} />}
-        />
+  const lightboxElements = [];
+
+  /**
+   * @param {string} defaultGroup
+   * @param {Element} element
+   * @param {number} index
+   */
+  function processLightboxElement(defaultGroup, element, index) {
```

Comment:
Any reason this is defined as an inner function? My concern is that it will be recreated on every time the outer function is called also while adding an event listener. Also, is there a `removeEventListener ` somewhere?

Reply:
It's because it adds to `lightboxElements` which is in this scope. There should only be one instance of this element if any on this document, so I'll refactor the code a bit assuming this.